views/js/ps_emailsubscription.js

```diff
@@ -0,0 +1,55 @@
+/**
+* 2007-2020 PrestaShop
+*
+* NOTICE OF LICENSE
+*
+* This source file is subject to the Academic Free License (AFL 3.0)
+* that is bundled with this package in the file LICENSE.txt.
+* It is also available through the world-wide-web at this URL:
+* http://opensource.org/licenses/afl-3.0.php
+* If you did not receive a copy of the license and are unable to
+* obtain it through the world-wide-web, please send an email
+* to license@prestashop.com so we can send you a copy immediately.
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+*  @author    PrestaShop SA <contact@prestashop.com>
+*  @copyright 2007-2020 PrestaShop SA
+*  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+*  International Registered Trademark & Property of PrestaShop SA
+*
+* Don't forget to prefix your containers with your own identifier
+* to avoid any conflicts with others containers.
+*/
+
+$(document).ready(function () {
+    $('.block_newsletter form').on("submit", function () {
+        if (psemailsubscription_subscription !== 'undefined') {
+            var psemailsubscriptionForm = $(this);
+            $('.block_newsletter_alert').remove();
+            $.ajax({
+                type: 'POST',
+                dataType: 'JSON',
+                url: psemailsubscription_subscription,
+                cache: false,
+                data: $(this).serialize(),
+                success: function (data) {
+                    if (data.nw_error) {
+                        psemailsubscriptionForm.prepend('<p class="alert alert-danger block_newsletter_alert">' + data.msg + '</p>');
```

Comment:
use `showErrorMessage` and `showSuccessMessage` instead? :thinking:

Reply:
Is only for bo, these are not available in front. I've stitched to non javascript behavior and alert messages in customer subscription/login.